protocol/x/subaccounts/keeper/subaccount.go

```diff
@@ -631,11 +642,52 @@ func (k Keeper) internalCanUpdateSubaccounts(
 			}
 		}
 
+		// Temporily apply open interest delta to perpetuals, so IMF is calculated based on open interest after the update.
+		// `perpOpenInterestDeltas` is only present for `Match` update type.
+		if perpOpenInterestDelta != nil {
+			if err := k.perpetualsKeeper.ModifyOpenInterest(
```

Comment:
Updating `perp.OpenInterest += delta` and then `perp.OpenInterest -= delta` technically modifies app hash as well. For `DeliverTx` this should be fine since this is called deterministically. For `CheckTx`, no need to keep each node `CheckState` the same